sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessibleByteArrayOutputStream.java

```diff
@@ -11,8 +11,34 @@
  * directly access the {@code byte[]} without copying.
  */
 public class AccessibleByteArrayOutputStream extends ByteArrayOutputStream {
+    /**
+     * Constructs an instance of {@link AccessibleByteArrayOutputStream} with the default initial capacity.
+     */
+    public AccessibleByteArrayOutputStream() {
+        super();
+    }
+
+    /**
+     * Constructs an instance of {@link AccessibleByteArrayOutputStream} with a specified initial capacity.
+     *
+     * @param initialCapacity The initial capacity.
+     * @throws IllegalArgumentException If {@code initialCapacity} is less than 0.
+     */
+    public AccessibleByteArrayOutputStream(int initialCapacity) {
+        super(initialCapacity);
+    }
+
     @Override
     public synchronized byte[] toByteArray() {
```

Comment:
does this have to be synchronized ? it's just returning reference.

Reply:
Probably not, this is an internal API so we can be smart about if it is used in a threaded scenario

Reply:
The apis in `ByteArrayOutputStream` seems to be synchronized because they make array copy when emitting return types, so they prevent against concurrent write. I think we can drop sychronized here which is yet another micro optimization.
If we want to be 100% correct then we should also make all "write" APIs throw after backing array reference is retrieved (which would make toByteArray kind of work like "close").